src/hotspot/share/ci/ciMethod.cpp

```diff
@@ -688,6 +688,11 @@ ciMethod* ciMethod::find_monomorphic_target(ciInstanceKlass* caller,
     return nullptr;
   }
 
+  // Redefinition support.
+  if (this->get_Method()->is_old() || root_m->get_Method()->is_old()) {
```

Comment:
Is it safe to access raw `Method*` from a compiler thread which is not in VM state?

Reply:
No, probably not.  I'll fix it.  I was assuming the whole function was in the VM state.